docs/how-it-works/hpo/explain-hpo.rst

```diff
@@ -0,0 +1,138 @@
+*    Bayesian Optimization: Uses a probabilistic surrogate model to learn the shape of the search space and predict promising configurations.
+*    Evolutionary Algorithms: Use search algorithms based on evolutionary concepts to evolve hyperparameter configurations, e.g. genetic programming.
+
+**Schedulers** manage the allocation of computational resources across multiple hyperparameter 
+configurations. They help prioritize promising configurations and terminate less effective 
+ones early to save resources. 
+
+Examples include:
+
+*    ASHA (Asynchronous Successive Halving Algorithm): Allocates resources by successively discarding the lowest-performing hyperparameter combinations.
+*    Median Stopping Rule: Stops trials that perform below the median of completed trials.
+
+The **evaluation metric** determines the performance of a model on a validation set. 
+Common metrics include accuracy, F1 score, and mean squared error. 
+The choice of metric depends on the task and its goals.
+
+A **trial** is the evaluation of one set of hyperparameters. Depending on whether you are 
+using a scheduler, this could be the entire training run, so as many epochs as you 
+have specified, or it could be terminated early and thus run for fewer epochs.
+
+
+When to Use HPO and Key Considerations
+---------------------------------------
+HPO can significantly enhance a model's predictive accuracy and generalization to unseen data 
+by finding the best hyperparameter settings.
+However, there are some drawbacks, especially with regards to computational cost and resource 
+management. Especially distributed HPO requires careful planning of computational resources 
+to avoid bottlenecks or excessive costs. You should consider that if you want to run four different trials, 
+and you run them on the same amount of resources as you normally would for only one set of hyperparameters, your training would run four times as long (unless you use a scheduler to elimate some trials early).
+
+Because of this, we want to design our HPO training wisely, so that we avoid unneseccary 
+computational cost. These are some things that might help you when you are getting started with HPO.
+
+HPO is beneficial when:
+
+*    model performance is sensitive to hyperparameter settings.
+*    you have access to sufficient computational resources.
+*    good hyperparameter settings are difficult to determine manually.
+
+To make sure you get the most out of your HPO training, the are some considerations you might want to make are to
+
+*    define the search space wisely: Narrow the search space to plausible ranges to improve efficiency.
+*    choose appropriate metrics: Use metrics aligned with your task's goals.
+*    allocate your resources strategically: Balance the computational cost with the expected performance gains.
+*    interpret your results carefully: Ensure that the improvements are meaningful and not due to overfitting.
```

Comment:
These last two are a bit vague IMO. Maybe add something short about how? E.g. ensuring that you won't overfit by using regularization, cross validation etc? And how does one balance the computation cost with the expected performance gains?

Reply:
Hmm good point! I will add some more info - it's a bit hard to balance and now make it into a ML tutorial, so I'll think about what could be useful to add :)

Reply:
Okay I added some more information to point two and three, and removed the last one (I think it's valid to say this is not HPO-specific enough to be mentioned here)

Reply:
Lmk if you still find it unclear :)

Reply:
Looks good now